modules/margin-protocol/src/lib.rs

```diff
@@ -796,6 +820,51 @@ impl<T: Trait> Module<T> {
 			Ok(())
 		}
 	}
+
+	/// Force closure position to liquidate liquidity pool based on opened positions.
+	///
+	/// Return `Ok` if closure success, or `Err` if not.
+	fn _liquidity_pool_close_position(pool: LiquidityPoolId, position_id: PositionId) -> DispatchResult {
+		let position = Self::positions(position_id).ok_or(Error::<T>::PositionNotFound)?;
+
+		let price = Self::_price(position.pair.base, position.pair.quote)?;
+		let spread = {
+			if position.leverage.is_long() {
+				T::LiquidityPools::get_bid_spread(pool, position.pair)
+					.ok_or(Error::<T>::NoBidSpread)?
+					.into()
+			} else {
+				T::LiquidityPools::get_ask_spread(pool, position.pair)
+					.ok_or(Error::<T>::NoAskSpread)?
+					.into()
+			}
+		};
+
+		let spread_profit = position
+			.leveraged_held
+			.checked_mul(&fixed_128_from_fixed_u128(price).saturating_mul(spread))
+			.ok_or(Error::<T>::NumOutOfBound)?;
+
+		let spread_profit_in_usd = Self::_usd_value(position.pair.base, spread_profit)?;
+		let penalty = spread_profit_in_usd;
+		let sub_amount = spread_profit_in_usd
+			.checked_add(&penalty)
+			.ok_or(Error::<T>::NumOutOfBound)?;
+
+		Self::_close_position(&position.owner, position_id, None)?;
+
+		let realized = cmp::min(
+			<T::LiquidityPools as LiquidityPools<T::AccountId>>::liquidity(position.pool),
+			u128_from_fixed_128(sub_amount),
```

Comment:
```suggestion
			u128_from_fixed_128(sub_amount.saturating_abs()),
```

sub_amount could be negative.